### PR TITLE
ClassObj

### PR DESCRIPTION
**Pull request by chandanghosh21**

### CODE REVIEW

File: min-max/min-max-sum/chandan-min-max-sum.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        for(var i=1; i<this.arr.length; i++) {  
+            if(max < this.arr[i]) {
+                max = this.arr[i];
+            }
+        }
+        return max;
+    }
+}
+
+class Sum extends MaxMin  {
```
Comment (kakanghosh) [inline min-max/min-max-sum/chandan-min-max-sum.js]: Change the class name, think something informative.

File: min-max/min-max-sum/chandan-min-max-sum.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    var sum = 0;
+    for(var i=0; i<this.arr.length; i++) {
+        sum += this.arr[i];
+     }
+     return sum;
+    }   
+}
+
+function getMaxOrMin(arr, option) {
+    var num = new Sum(arr);
```
Comment (kakanghosh) [inline min-max/min-max-sum/chandan-min-max-sum.js]: can you explain how `arr` parameter is assigning to the class `this.arr` variable?

File: min-max/min-max-sum/chandan-min-max-sum.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    getMax() {
+        var max = this.arr[0];
+        for(var i=1; i<this.arr.length; i++) {  
+            if(max < this.arr[i]) {
+                max = this.arr[i];
+            }
+        }
+        return max;
+    }
```
Comment (kakanghosh) [inline min-max/min-max-sum/chandan-min-max-sum.js]: add `getter` and `setter` for `arr` property.

File: min-max/min-max-sum/chandan-min-max-sum.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            value = num.getMax();
+        break;
+        case 'MIN':
+            value = num.getMin();
+        break;
+        case 'SUM':
+            value = num.getSum();
+        break;
+        default:
+            value = "Invalid option. Provider 'MIN' or 'MAX' or 'SUM'";
```
Comment (kakanghosh) [inline min-max/min-max-sum/chandan-min-max-sum.js]: Typo: `provider`

File: min-max/min-max-sum/chandan-min-max-sum.js (hunk 2 above)
Comment (chandanghosh21) [reply min-max/min-max-sum/chandan-min-max-sum.js]: as sum class extended the MaxMin class it's automatically getting the array or should I set the constructor using super keyword?

File: min-max/min-max-sum/chandan-min-max-sum.js (hunk 3 above)
Comment (chandanghosh21) [reply min-max/min-max-sum/chandan-min-max-sum.js]: would you please explain a bit?

File: min-max/min-max-sum/chandan-min-max-sum.js (hunk 4 above)
Comment (chandanghosh21) [reply min-max/min-max-sum/chandan-min-max-sum.js]: Copy paste effects.

**Pull request merged**